src/Libplanet.Net/Messages/GetBlockHashesMsg.cs

```diff
@@ -1,48 +1,26 @@
-using System;
 using System.Collections.Generic;
-using System.Linq;
 using Libplanet.Blockchain;
 using Libplanet.Types.Blocks;
 
 namespace Libplanet.Net.Messages
 {
     internal class GetBlockHashesMsg : MessageContent
     {
-        public GetBlockHashesMsg(BlockLocator locator, BlockHash? stop)
```

Comment:
Is it backward compatible?

Reply:
The backward compatibility has been broken since #3913, albeit on a different level. This further(?) breaks the compatibility since it has been already broken. 🙄

Reply:
The result is the same, any node running with a different Libplanet version would not be able to sync to a node running the newest version.

Reply:
So is there any way to test a new version of Libplanet on a network where the live version is running?

Reply:
It is possible to apply only the business logic **and force partial compatibility** as I have with #3913, but I don't think that is necessary (or worth the time). If you are still worried, we can further divide this PR into multiple releases where

- We keep partial compatibility.
  - Message level compatibility is left intact.
  - Ignore `BlockLocator`s with multiple `BlockHash`es and/or non-`null` `Stop`.
  - This results in a newer node being able to sync to an old node, but not the other way around.
- We break compatibility completely by cleaning up/updating the `GetBlockHashesMsg` encoding.
  - This would result in a newer node and an old node not being able to sync to each other.

As the newer node is about getting rid of "unnecessary" block sync logic and dependencies, the compatibility has to be broken at some point.

Reply:
I don't think backward compatibility is necessary on this. But, at least, we need to test this feature on our dev cluster.

Reply:
Arg, might as well separate the encoding part into a separate PR for easier testing. 😕